### PR TITLE
tweak creating the replace cand set to have more possible options

### PR DESCRIPTION
**Pull request by snuffysasa**
tweak creating the replace cand set to have more possible options

### CODE REVIEW

File: src/randomizer.py [hunk 1]
```diff
@@ -712,20 +712,22 @@ def find_duplicates(e: Expression) -> None:
     else:
         replace_subexprs(fn.body, find_duplicates)
 
-    assert orig_expr in replace_cands
```
Comment (simonlindholm) [inline src/randomizer.py:715]: worth keeping this assertion I think

File: src/randomizer.py [hunk 2]
```diff
@@ -712,20 +712,22 @@ def find_duplicates(e: Expression) -> None:
...
+        if random_bool(random, 0.5):
+            replace_cand_set.update(replace_cands[: index + 1])
+        else:
+            replace_cand_set.update(replace_cands[index:])
+    else:
+        replace_cand_set.add(orig_expr)
+
+    for cand in replace_cands:
+        if random_bool(random, 0.5):
+            replace_cand_set.add(cand)
```
Comment (simonlindholm) [inline src/randomizer.py:730]:
```suggestion
    if random_bool(random, 0.5):
        for cand in replace_cands:
            if random_bool(random, 0.5):
                replace_cand_set.add(cand)
```
to prioritize more natural replacement ranges

**Review by simonlindholm — APPROVED** [review]

File: src/randomizer.py (hunk 1 above)
Comment (snuffysasa) [reply src/randomizer.py:715]: I had this moved to a different spot, which made sense / was necessary for a different PR I was working on.

But I'll move it back for now, and can discuss moving it for the other PR later

File: src/randomizer.py (hunk 2 above)
Comment (snuffysasa) [reply src/randomizer.py:730]: yea this makes sense

**Pull request merged**